<launch>
  <arg name="USE_LASER" default="true" />
  <arg name="USE_SCREENPOINT" default="false" />
  <arg name="USE_VIEW" default="false" />
  <arg name="USE_LIMB_TRAJECTORY" default="true" />

  <arg name="USE_IMAGE_TRANSPORT" default="false" />
  <arg name="USE_OCTOMAP" default="false" />
  <arg name="USE_WEB" default="false" />
  <arg name="USE_INTERACTIVE_JOINT" default="false" />
  <arg name="USE_INTERACTIVE_ENDCOORDS" default="false" />

  <arg name="USE_LOW_BANDWIDTH" default="false" />
  
  <!-- when we use lower controller -->
  <arg name="USE_CONTROLLER" default="true" />
  <group if="$(arg USE_CONTROLLER)" >
    <env name="HRPSYS_GAZEBO_ATLAS_COMMAND_TOPIC_NAME" value="/atlas/atlas_command_hrpsys" />
    <include file="$(find hrpsys_gazebo_atlas)/launch/atlas_touchit_server.launch" />
  </group>
  <group unless="$(arg USE_CONTROLLER)" >
    <include file="$(find hrpsys_gazebo_atlas)/launch/atlas_ik.launch" /> 
  </group>
  
  <!-- obsolated -->
  <arg name="USE_HAND"  default="false" />
  <arg name="USE_BDI_ACTION" default="false" />

  <!-- hrpsys controller launch -->
  <include file="$(find hrpsys_gazebo_atlas)/launch/atlas_hrpsys_bringup.launch">
    <arg name="USE_LIMB_TRAJECTORY" value="$(arg USE_LIMB_TRAJECTORY)" />
  </include>

  <!-- ROS nodes -->
  <!-- laser scanning -->
  <include if="$(arg USE_LASER)"
           file="$(find hrpsys_gazebo_atlas)/launch/atlas_laser.launch" />

  <!-- screen point -->
  <include if="$(arg USE_SCREENPOINT)"
           file="$(find hrpsys_gazebo_atlas)/launch/atlas_screenpoint.launch" >
    <arg name="USE_VIEW" value="$(arg USE_VIEW)" />
  </include>

  <!-- image transport -->
  <include if="$(arg USE_IMAGE_TRANSPORT)"
           file="$(find hrpsys_gazebo_atlas)/launch/atlas_imagetransport.launch" />

  <!-- octomap -->
  <include if="$(arg USE_OCTOMAP)"
           file="$(find hrpsys_gazebo_atlas)/launch/atlas_octomap.launch" >
    <arg name="USE_TILT_LASER" value="false" />
  </include>

  <!-- hand controller -->
  <include if="$(arg USE_HAND)"
           file="$(find hrpsys_gazebo_atlas)/launch/sandia_hand_controller.launch" />

  <!-- bdi_controller -->
  <node if="$(arg USE_BDI_ACTION)"
        pkg="atlas_msgs" type="actionlib_server" name="walking_actionlib_server" />

  <!-- tf for sensors -->
  <include file="$(find hrpsys_gazebo_atlas)/launch/atlas_sensor_tf.launch" />

  <!-- web UI -->
  <include if="$(arg USE_WEB)"
	   file="$(find hrpsys_gazebo_atlas)/launch/atlas_web.launch" />


  <!-- compress point cloud and joint states -->
  <include if="$(arg USE_LOW_BANDWIDTH)"
	   file="$(find hrpsys_gazebo_atlas)/launch/atlas_low_bandwidth_ocs.launch"/>

  <!-- interactive joint marker and rviz -->
  <include file="$(find hrpsys_gazebo_atlas)/launch/atlas_interactive_marker.launch" if="$(arg USE_INTERACTIVE_JOINT)" >
    <arg name="USE_INTERACTIVE_JOINT" value="$(arg USE_INTERACTIVE_JOINT)"/>
    <arg name="USE_INTERACTIVE_ENDCOORDS" value="$(arg USE_INTERACTIVE_ENDCOORDS)"/>
  </include>

  <node pkg="hrpsys_gazebo_atlas" type="atlas-motion-sequence-player.l" name="atlas_motion_sequence_player" output="log" />

</launch>
